### ISSUE TITLE
Inconsistent Add Template workflow for Pages/Single Posts

### ISSUE THREAD

**Issue by bgardner** [WordPress/gutenberg#55056]
### Description

I have a theme that includes both single.html and page.html files. When using the Add Template feature, I am met with difference experiences:

**Page** - Only option is to create template that is applied to a specific item (as seen below)

![image](https://github.com/WordPress/gutenberg/assets/486261/43dcd866-6e50-4a6b-b63c-f49b5c4ea0e2)

**Singe item: Post** - Options include creating a template that is applied to a specific item AND for all items (as seen below)

![image](https://github.com/WordPress/gutenberg/assets/486261/1a944f60-19c8-4c35-a4dc-ad3059c9b4ae)

Worth noting that removing the page.html file from the theme allows both options upon Page template creation. Obviously this is not ideal (and that requirement would be inconsistent with Single Posts).

### Step-by-step reproduction instructions

1. Go to Site Editor > Templates +
2. Compare process of adding Page and Single Posts template.
3. Or see video below.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/486261/5329a8ff-1019-4df0-9731-c9c601f8e380

### Environment info

WordPress 6.4-beta2, with or without Gutenberg active.
WordPress 6.3.1, issue also exists, with or without Gutenberg active.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by richtabor** [comment]
I agree that these paths should be treated the same, pages and posts.
